### PR TITLE
remove reflection from core

### PR DESCRIPTION
**Pull request by jafu888**

### CODE REVIEW

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/WidgetFrame.java [hunk 1]
```diff
@@ -419,6 +420,10 @@ public boolean setValue(String key, CLElement value) throws CLParsingException {
         return true;
     }
 
+    public String getId() {
+        return widget.stringId;
```
Comment (oscar-ad) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/WidgetFrame.java:424]: What if the widget is null?

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java [hunk 2]
```diff
@@ -420,7 +448,7 @@ private static int clamp(int c) {
         return c;
     }
 
-    public void setInterpolatedValue(View view, float[] value) {
+    public void setInterpolatedValue_(View view, float[] value) {
```
Comment (oscar-ad) [inline constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java:451]: Is this intentional?

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/motion/MotionWidget.java [hunk 3]
```diff
@@ -458,4 +459,8 @@ public void setCustomAttribute(String name, int type, String value) {
     public CustomVariable getCustomAttribute(String name) {
         return widgetFrame.getCustomAttribute(name);
     }
+    public void setInterpolatedValue(CustomAttribute attribute, float[] mCache) {
```
Comment (oscar-ad) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/motion/MotionWidget.java:462]: Add a line between these

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/WidgetFrame.java [hunk 4]
```diff
@@ -587,4 +599,6 @@ void logv(String str) {
         System.out.println(ss + " " + str);
     }
 
+    public void setCustomValue(CustomAttribute valueAt, float[] mTempValues) {
+    }
```
Comment (oscar-ad) [inline constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/WidgetFrame.java:603]: unused?

**Review by oscar-ad — CHANGES_REQUESTED** [review]

File: constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/WidgetFrame.java (hunk 4 above)
Comment (jafu888) [reply constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/WidgetFrame.java:603]: Called but not implemented.  This is for custom Attribute support in the Compose

File: constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java (hunk 2 above)
Comment (jafu888) [reply constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintAttribute.java:451]: Removed

**Review by oscar-ad — APPROVED** [review]

**Pull request merged**